openretina/data_io/cyclers.py

```diff
@@ -36,6 +36,10 @@ def __init__(self, loaders: dict[str, DataLoader], shuffle: bool = True):
         self.shuffle = shuffle
 
     def __iter__(self):
+        worker_info = torch.utils.data.get_worker_info()
+        if worker_info is not None:
+            raise NotImplementedError("LongCycler does not support multiple workers yet.")
```

Comment:
We could allow num_workers == 1. But this also lead to slower training times, right?

Reply:
I did not test this, but I cannot see how it could be faster yes

Reply:
Theoretically, because the two processes (for training on gpu and for data loading) can run in parallel. But you're right, based on the benchmarks it seems the overhead is greater than the gain.

Reply:
Hm, I changed my mind, let's allow one worker, too (I tried it and it works and speed is similar):
```suggestion
        worker_info = torch.utils.data.get_worker_info()
        if worker_info is not None and worker_info.num_workers > 1:
            raise NotImplementedError("LongCycler does not support multiple workers yet.")
```